mountpoint-s3-fs/tests/fuse_tests/metrics_test.rs

```diff
@@ -0,0 +1,226 @@
+use crate::common::fuse::{TestSessionConfig, mock_session, read_dir_to_entry_names};
+use crate::common::test_recorder::{Metric, TestRecorder};
+use mountpoint_s3_fs::S3FilesystemConfig;
+use mountpoint_s3_fs::metrics::defs::{
+    ATTR_FUSE_REQUEST, FUSE_IO_SIZE, FUSE_REQUEST_ERRORS, FUSE_REQUEST_LATENCY, PREFETCH_RESET_STATE,
+};
+use rusty_fork::rusty_fork_test;
+use std::fs::{File, read_dir};
+use std::io::{Read, Seek, Write};
+use std::sync::Arc;
+
+fn setup_recorder() -> TestRecorder {
+    let recorder = TestRecorder::default();
+    metrics::set_global_recorder(recorder.clone()).expect("Failed to set global recorder");
+    recorder
+}
+
+fn get_metric(recorder: &TestRecorder, name: &str, labels: &[(&str, &str)]) -> Arc<Metric> {
+    recorder
+        .get(name, labels)
+        .unwrap_or_else(|| panic!("Expected metric '{name}' with labels {labels:?} to exist"))
+}
+
+fn get_histogram(recorder: &TestRecorder, name: &str, labels: &[(&str, &str)]) -> Vec<f64> {
+    let metric = get_metric(recorder, name, labels);
+    match metric.as_ref() {
+        Metric::Histogram(_) => metric.histogram(),
+        _ => panic!("Metric '{name}' is not a histogram"),
+    }
+}
+
+fn assert_metric_exists(recorder: &TestRecorder, name: &str, labels: &[(&str, &str)]) {
+    let _ = get_metric(recorder, name, labels);
+}
```

Comment:
Not blocking but it would be good to update at some point. These should be macros in order to preserve the line numbers when an assertion failed. Otherwise, we only get the line numbers here (21 and 28) for each test failure.

As an example, here's what we do for permission assertions elsewhere: https://github.com/awslabs/mountpoint-s3/blob/5502a861ee11eaa6dc61aa8e711262b2d4388d8c/mountpoint-s3-fs/src/data_cache/cache_directory.rs#L170-L182

It would help for instance with the stack trace here: https://github.com/awslabs/mountpoint-s3/pull/1653#pullrequestreview-3356705006